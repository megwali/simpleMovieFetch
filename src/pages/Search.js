import { Link } from 'react-router-dom';
import styled from 'styled-components';

const ResultContainer = styled.div`
  display: flex;
  flex-flow: row wrap;
  max-width: 100%;
`;

const SearchContainer = styled.div`
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
  margin: 1.5rem;
  width: 100%;
`;

const SearchInput = styled.input`
  margin-bottom: 1.5rem;
  padding: 0.5rem;
  width: 50%;
`;

const StyledLink = styled(Link)`
  min-width: 20%;
`;

const Thumbnail = styled.img`
  margin-bottom: 2rem;
  margin-right: 2rem;
`;

const Search = ({ onChange, onClick, results }) => (
  <SearchContainer>
    <SearchInput onChange={({ target }) => onChange(target.value)} type="search" />

    <ResultContainer>
      {results?.map(({ show }) => (
        <StyledLink key={show.id} onClick={() => onClick(show)} to="/show">
          <Thumbnail alt={show?.name} src={show?.image?.medium} />
        </StyledLink>
      ))}
    </ResultContainer>
  </SearchContainer>
);

export default Search;
