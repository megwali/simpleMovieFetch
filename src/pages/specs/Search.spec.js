import { create } from 'react-test-renderer';
import Search from '../Search';

describe('Search', () => {
  it('renders', () => {
    const component = create(<Search />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
