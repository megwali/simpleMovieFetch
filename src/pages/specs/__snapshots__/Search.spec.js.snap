// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders 1`] = `
<div
  className="sc-gsTCUz hlZFEK"
>
  <input
    className="sc-dlfnbm RbZbw"
    onChange={[Function]}
    type="search"
  />
  <div
    className="sc-bdfBwQ iWUBct"
  />
</div>
`;
